src/InlineRelationsTransformer.cpp

```diff
@@ -630,9 +631,10 @@ NullableVector<AstArgument*> getInlinedArgument(AstProgram& program, const AstAr
                 }
             }
         } else if (const auto* udf = dynamic_cast<const AstUserDefinedFunctor*>(arg)) {
-            for (size_t i = 0; i < udf->getArity(); i++) {
+            for (size_t i = 0; i < udf->getArguments().size(); i++) {
```

Comment:
can you change this loop to size_t i=0; for(auto arg: fun.getArguments()) { ....; i++}